operator/src/main/java/dev/responsive/k8s/operator/reconciler/KafkaStreamsPolicyPlugin.java

```diff
@@ -101,13 +104,80 @@ public void reconcile(
     final var targetState = maybeTargetState.get();
     LOG.info("target state for app {} {}", appName, targetState);
 
-    if (targetState.getTargetState().isEmpty()) {
+    if (targetState.getTargetState().isPresent()) {
       LOG.info(
           "we were not able to get a target state from controller, so don't try to reconcile one");
-      return;
+      maybeScaleApplication(
+          targetState.getTargetState().get().getKafkaStreamsState().getReplicas(),
+          managedApp,
+          appNamespace,
+          appName,
+          ctx
+      );
+    }
+
+    for (final Action action : targetState.getActions()) {
+      switch (action.getActionCase()) {
```

Comment:
does it make sense for the action to implement the action itself instead of having a big switch statement? e.g.
```
targetState.getActions().forEach(action -> action.execute(...))
```
That would make it easier to review each action state

Reply:
This is the type generated by protobuf, so we can't customize it like that. We'd need to build up a mirror set of types for use in the main code, and then do conversions back and forth when serializing/deserializing. I do like this approach in general, since these proto-generated types are annoying to deal with. But that's a bigger change than I want to take on here.